pymeasure/instruments/resources.py

```diff
@@ -58,3 +61,24 @@ def list_resources():
             print(e)
     rm.close()
     return instrs
+
+
+def find_serial_port(vendor_id: Optional[int] = None, product_id: Optional[int] = None,
+                     serial_number: Optional[str] = None) -> str:
+    """Find the VISA port name of the first serial device with the given USB information.
+
+    Use `None` if you do not want to check for that property.
+
+    :param int vid: Vendor ID.
+    :param int pid: Product ID.
+    :param str sn: Serial number.
+    :return: Port as a VISA string for a serial device (e.g. "ASRL5" or "ASRL/dev/ACMO5").
+    """
+    for port in sorted(list_ports.comports()):
+        if ((vendor_id is None or port.vid == vendor_id)
+                and (product_id is None or port.pid == product_id)
+                and (serial_number is None or port.serial_number == str(serial_number))):
+            # remove "COM" from windows serial port names.
+            port_name = port.device.replace("COM", "")
+            return "ASRL" + port_name
+    raise AttributeError("No device found for the given data.")
```

Comment:
I am not 100% sure, but I think that the association between the COM number and ASRL number is not always reliable. I mean that COM4 can be ASRL5.

Reply:
Then how do you know the VISA name of a device?
Afaik, for each serial port, there is an ASRLport with that name (so ASRL5 for the **windows** COM5 or ASRL/dev/ACM0 for /dev/ACM0), but that the backend can introduce additional aliases, such that NI has the **VISA** COM5 linked to ASRL5.

Reply:
You can give a VISA name "COM4" to the VISA port "ASRL5" which is linked to the fifth windows COM port (COM5), but the connection between the com-Port and ASRL5 is fixed.
At least in NI-MAX it was shown to be linked, but an alias (defaulting to "COM5") can be set.

Reply:
@bmoneke,thanks for the clarification. I remember having problems with this association in the past, but it could have been something wrong on my side.